src/Sparrow/Json/LazyNumberValue.cs

```diff
@@ -161,6 +161,19 @@ private static void ThrowInvalidNumberFormat(LazyNumberValue self, string type)
             return -(decimal)y;
         }
 
+        public static bool operator !=(LazyNumberValue self, LazyNumberValue lnv) => !(self == lnv);
+        
+        public static bool operator ==(LazyNumberValue self, LazyNumberValue lnv)
+        {
+            if (ReferenceEquals(self, null) && ReferenceEquals(lnv, null))
```

Comment:
isn't it better to write

```csharp
            if (ReferenceEquals(self, null) == false)
            {
                if (ReferenceEquals(lnv, null))
                    return false;

                return self.Equals(lnv);
            }

            if (ReferenceEquals(lnv, null))
                return true;

            return lnv.Equals(self);
```

less checks?

Reply:
I wanted to make this similar to `==` operator for `LazyStringValue`, but I think we can use your code instead.